### DISCUSSION TITLE
app.param() equivalent?

### DISCUSSION THREAD

**Discussion by kjjd84** [honojs/hono#3445] [Q&A]
how do we do route model binding in hono?

express example: https://www.codementor.io/@mirko0/express-js-route-model-binding-wris4hd2m

**Comment by safwanyp** [comment]
Based on the docs, there is no 1:1 equivalent to this. 
However, you might be able to create a middleware, and do something like this:
```js
const idMw = createMiddleware(async (c, next) => {
  const id = 1029;
  c.set("id", id);
  await next();
});

app.use("/api/:id", idMw);

app.get("/api/:id", async c => {
  const id = c.get("id");

  return c.json({ id });
});
```
